include/mesh.h

```diff
@@ -497,6 +498,8 @@ class Mesh {
       particle_velocity_constraints_;
   //! Vector of generators for particle injections
   std::vector<mpm::Injection> particle_injections_;
+  //! Nodal property pool
+  mpm::NodalProperties nodal_properties_;
```

Comment:
Why are we are storing this as an object, rather than a pointer?

Reply:
Should I change it to a shared_ptr?